common/grammar-parser.cpp

```diff
@@ -137,6 +152,87 @@ namespace grammar_parser {
             bool                                 is_nested) {
         size_t last_sym_start = out_elements.size();
         const char * pos = src;
+
+        auto handle_repetitions = [&](int min_times, int max_times) {
```

Comment:
I'm on the fence about doing this as a lambda (is that what this would be called in C?) -- to be consistent with the rest of the file, it seems like this would be a static function named `parse_repetitions`? I'm guessing it's not as convenient because one would have to pass in a ton of parameters and whatnot, but I'm curious to hear what others think. @ggerganov would appreciate you smelling this one and giving us your guidance.

Reply:
All the parse methods advance the char pointer and return it. This one doesn’t “parse”, just fiddles with the out_elements array. Lambdas are efficient and make for much simpler code (precisely avoiding that ton of params), one of the few reasons to not ditch C++ just yet imho